gateleen-core/src/main/java/org/swisspush/gateleen/core/http/LocalHttpServerResponse.java

```diff
@@ -216,7 +216,7 @@ public Future<Void> write(Buffer data) {
         if (!chunked && !headers.contains(HttpHeaders.CONTENT_LENGTH)) {
             IllegalStateException ex = new IllegalStateException("You must set the Content-Length header to be the total size of the message "
                     + "body BEFORE sending any data if you are not using HTTP chunked encoding.");
-            logger.error("non-proper HttpServerResponse occured", ex);
+            logger.debug("non-proper HttpServerResponse occured", ex);
```

Comment:
Shouldn't we use the _exceptionFactory_ here too? How does changing the log level ensure that the error is only logged once?

Reply:
> How does changing the log level ensure that the error is only logged once?

I does not directly. Instead the duplication only exists as soon debug logging is active.